faster_whisper/transcribe.py

```diff
@@ -661,6 +669,32 @@ def generate_with_fallback(
 
             if not needs_fallback:
                 break
+        else:
+            # all failed
+            filtered_results = {}
+            avg_logprob_index = 1
+            compression_ratio_index = 3
+
+            # filter out results with compression ratio below compression_ratio_threshold
+            for key, result_data in results.items():
+                if (
+                    options.no_speech_threshold is not None
+                    and result_data[compression_ratio_index]
+                    < options.compression_ratio_threshold
+                ):
+                    filtered_results[key] = result_data
+
+            # select the optimal result based on the maximum avg_logprob
+            if filtered_results:
+                result, avg_logprob, final_temperature, compression_ratio = max(
```

Comment:
Let's avoid unpacking the values to avoid repeating the variables multiple times.

Similar to the function `decode_with_fallback` in openai/whisper, you can save the result in a tuple named for example `decode_result` and return this tuple directly.